drivers/shared/executor/executor_test.go

```diff
@@ -367,3 +392,73 @@ func setupRootfsBinary(t *testing.T, rootfs, path string) {
 	err = os.Link(src, dst)
 	require.NoError(t, err)
 }
+
+func TestExecutor_Start_Kill_Immediately_NoGrace(pt *testing.T) {
+	pt.Parallel()
+	for name, factory := range executorFactories {
+		pt.Run(name, func(t *testing.T) {
+			require := require.New(t)
+			execCmd, allocDir := testExecutorCommand(t)
+			execCmd.Cmd = "/bin/sleep"
+			execCmd.Args = []string{"100"}
+			factory.configureExecCmd(t, execCmd)
+			defer allocDir.Destroy()
+			executor := factory.new(testlog.HCLogger(t))
+			defer executor.Shutdown("", 0)
+
+			ps, err := executor.Launch(execCmd)
+			require.NoError(err)
+			require.NotZero(ps.Pid)
+
+			waitCh := make(chan interface{})
+			go func() {
+				defer close(waitCh)
+				executor.Wait(context.Background())
+			}()
+
+			require.NoError(executor.Shutdown("SIGKILL", 0))
+
+			select {
+			case <-waitCh:
+				// all good!
+			case <-time.After(4 * time.Second * time.Duration(tu.TestMultiplier())):
+				require.Fail("process did not terminate despite SIGKILL")
+			}
+		})
+	}
+}
+
+func TestExecutor_Start_Kill_Immediately_WithGrace(pt *testing.T) {
+	pt.Parallel()
+	for name, factory := range executorFactories {
+		pt.Run(name, func(t *testing.T) {
+			require := require.New(t)
+			execCmd, allocDir := testExecutorCommand(t)
+			execCmd.Cmd = "/bin/sleep"
+			execCmd.Args = []string{"100"}
+			factory.configureExecCmd(t, execCmd)
+			defer allocDir.Destroy()
+			executor := factory.new(testlog.HCLogger(t))
+			defer executor.Shutdown("", 0)
+
+			ps, err := executor.Launch(execCmd)
+			require.NoError(err)
+			require.NotZero(ps.Pid)
+
+			waitCh := make(chan interface{})
+			go func() {
+				defer close(waitCh)
+				executor.Wait(context.Background())
+			}()
+
+			require.NoError(executor.Shutdown("SIGKILL", 100*time.Millisecond))
```

Comment:
We should probably either change the test to assert the grace period has some effect or remove the test. Right now it only asserts that specifying a non-zero timeout doesn't return an error which doesn't seem very useful.

Reply:
I'd rather keep the test, the test is still valuable in that we are testing that processes do die eventually in all permutations of executor and grace paths.  While we can fine tune and add more assertions, this seems like an ok baseline to keep.